scripts/build.sh

```diff
@@ -1,31 +1,10 @@
-#!/bin/bash
-set -e
-set -x
-BASE="`dirname $0`"
-
-mkdir -p $BASE/../lib
-mkdir -p $BASE/../bin
-
-CFLAGS="--std=c++17 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0"
-INCLUDE=(
-  "-I/cvmfs/cms.cern.ch/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/include" # ROOT
-  "-I/cvmfs/cms.cern.ch/slc7_amd64_gcc820/external/boost/1.72.0-bcolbf/include" # BOOST
-)
-LIB=(
-  "-L/cvmfs/cms.cern.ch/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_1_0_pre6/external/slc7_amd64_gcc820/lib"
-)
-ROOTLIBS=(-lCore -lRIO -lNet -lHist -lMatrix -lThread -lTree -lMathCore -lGpad -lGraf3d -lGraf -lPhysics -lPostscript -lASImage)
-OTHERLIBS=(-ldl -ljpeg -lpng15 "-lstdc++fs")
-# I have no idea why G++ does not read symbols from here with -lstdc++fs. I got that filename out of strace, so it was sure opened...
-HACKS="/cvmfs/cms-ib.cern.ch/nweek-02654/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../lib64/libstdc++fs.a" 
-
-# Compile
-g++ $CFLAGS "${INCLUDE[@]}" -c $BASE/../src/DQMRenderPlugin.cc -o $BASE/../lib/DQMRenderPlugin.o
-g++ $CFLAGS "${INCLUDE[@]}" "${OTHERLIBS[@]}" -c $BASE/../src/render.cc -o $BASE/../lib/render.o
-g++ --shared "${LIB[@]}" "${ROOTLIBS[@]}" "${OTHERLIBS[@]}" $BASE/../lib/DQMRenderPlugin.o -o $BASE/../lib/renderplugin.so
-
-# Link executable
-g++ $CFLAGS "${LIB[@]}" "${ROOTLIBS[@]}" "${OTHERLIBS[@]}" $BASE/../lib/renderplugin.so $BASE/../lib/render.o -o $BASE/../bin/render $HACKS
-
-# Link render plugins
-g++ -O3 $CFLAGS "${INCLUDE[@]}" --shared $BASE/../plugins/*.cc -o $BASE/../lib/libDQMRenderPlugins.so
+#!/bin/bash -e
+CMSSW_VERSION=CMSSW_11_1_0_pre6
```

Comment:
Hi @smuzaffar, I see that CMSSW version is hardcoded in the build script. Does it mean that we only have to change this value in order to get the dependencies from a different CMSSW version?

Reply:
fopr building dqmgui RPM, I am not going to use build.sh script. Thsi is just for you so that you can build locally. You can change the CMSSW version and will get the deps from that version.

For building RPMs (via cmsdist) , I am going to directly run `gmake` and root/boost and other dependencies will automatically picked up from the cmsdist specs.

Reply:
Sorry for my lack of understanding but I don't quite get yet how can we choose the CMSSW version that will provide the dependencies for the DQM GUI when the DQM GUI will be deployed using CMSDIST. Could you elaborate a bit more on this please?

Reply:
dqmqui does not depend on cmssw version, it depends on the root version used by cmssw ... right. For when we build Releases for CMSSW_11_2_X, then we will also build dqmgui along with it. dqmgui will use the same root version as cmssw has used. 

When you will deploy dqmgui then you will only get dqmgui and its dependencies (e.g. root , python packages), You are not going to get a full cmssw release along with dqmgui

Reply:
Thanks, now it's more clear! So CMSDIST will have a list of CMSSW releases with respective DQM GUI releases associated with each of them. Then, the only thing that we (DQM) have to do is to ask CMSDIST to provide a DQM GUI that is associated with a concrete CMSSW release. If my reasoning is correct, could you provide a code snippet for us to do that?

Reply:
You mean you want to know what is the dqmgui tag/package for a gievn cmssw release? If yes then sure I can provide you a way to find it. It is not there yet but easy to add.

Reply:
We frequently have to deploy new/old CMSSW versions. What we would like to do, is to **automatically** deploy the DQM GUI, at the same time. So a perfect scenario would be to have a command that takes CMSSW version as an argument and returns the DQM GUI that was build with the dependencies associated with the CMSSW version that was provided as an argument.

Reply:
sure, I understand. I will provide you such command when ready